Project/DAQ_System/Program/Src/app.cpp

```diff
@@ -117,22 +116,265 @@ void cppMain() {
 		displacement_inches = lin_pot->DisplacementInches();
 
 
-		printf("\n Percentage: %f", displacement_inches);
+		//printf("\n Percentage: %f", displacement_inches);
+
 		if(CANMessageReceivedFlag){
 			__disable_irq();
-			TxData[0] = 1 ;
-			TxData[1] = 2 ;
-			TxData[2] = 3 ;
-			TxData[3] = RxData[3] ;
-			TxData[4] = RxData[4] ;
-			TxData[5] = 6 ;
-			TxData[6] = 0x12 ;
-			TxData[7] = (RxData[4] * 256 + RxData[3]) * .1;
-
-			HAL_CAN_AddTxMessage(&hcan1, &TxHeader,TxData,&TxMailbox);
-			CANMessageReceivedFlag = false;
+			canParseDataFrame();
+			if(CanID == 0x0CFFF048){
+				printf("\t\t 0x0CFFF048 \n");
+				printf("\r");
+				printf("RPM = %.2f ", DecodeDataFrame[0]);
+				printf("TPS = %.2f ", DecodeDataFrame[2]);
+				printf("Fuel Open Time = %.2f ", DecodeDataFrame[4]);
+				printf("Ignition Angle = %.2f \n", DecodeDataFrame[6]);
+				printf("\r");
+			}
+			if(CanID == 0x0CFFF548){
+				printf("\t\t 0x0CFFF548 \n");
+				printf("\r");
+				printf("Battery Volt = %.2f ", DecodeDataFrame[0]);
+				printf("Air Temp = %.2f ", DecodeDataFrame[2]);
+				printf("Coolant Temp = %.2f ", DecodeDataFrame[4]);
+				printf("Temp Type = %.2f \n", DecodeDataFrame[6]);
+				printf("\r");
+			}
 			__enable_irq();
 
 		}
 	}
 }
+
+
+void canFilterSetup(int FilterBankNum, uint32_t FilterExtID ){
+
+	// Filter initialization
+	CAN.FilterIdHigh = FilterExtID >> 13 ;
+	CAN.FilterIdLow = ((FilterExtID & 0x1FFF) <<3) | CAN_ID_EXT | CAN_RTR_DATA ;
+	CAN.FilterMaskIdHigh = 0xFFFF;
```

Comment:
in the future i would like to actually configure a proper filter mask. It would be something like 0x00001100 Where the 1 represent hex digits that do get compared and the 0s are the hex digits that are don't cares